From 24d12413eca8f59082a9a39db562a16abce47738 Mon Sep 17 00:00:00 2001
From: Brian McGillion <bmg.avoin@gmail.com>
Date: Tue, 28 Oct 2025 07:37:59 +0400
Subject: [PATCH] hostnamed, localed, timedated: re-enable methods that change
 system settings

This patch re-enables the system settings modification methods that were
disabled by the NixOS patch. This allows proper locale and keyboard
configuration management in Ghaf systems.

Signed-off-by: Gabriel Ebner <gebner@gebner.org>
Signed-off-by: Santtu Lakkala <santtu.lakkala@unikie.com>
Signed-off-by: Brian McGillion <bmg.avoin@gmail.com>
---
 src/locale/localed.c | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/src/locale/localed.c b/src/locale/localed.c
index dd4d9e59ea..11d9130bbc 100644
--- a/src/locale/localed.c
+++ b/src/locale/localed.c
@@ -221,9 +221,6 @@ static int method_set_locale(sd_bus_message *m, void *userdata, sd_bus_error *er

         use_localegen = locale_gen_check_available();

-        return sd_bus_error_setf(error, SD_BUS_ERROR_NOT_SUPPORTED,
-            "Changing system settings via systemd is not supported on NixOS.");
-
         /* If single locale without variable name is provided, then we assume it is LANG=. */
         if (strv_length(l) == 1 && !strchr(l[0], '=')) {
                 if (!locale_is_valid(l[0]))
@@ -341,9 +338,6 @@ static int method_set_vc_keyboard(sd_bus_message *m, void *userdata, sd_bus_erro
         if (r < 0)
                 return bus_log_parse_error(r);

-        return sd_bus_error_setf(error, SD_BUS_ERROR_NOT_SUPPORTED,
-            "Changing system settings via systemd is not supported on NixOS.");
-
         vc_context_empty_to_null(&in);

         r = vc_context_verify_and_warn(&in, LOG_ERR, error);
@@ -461,9 +455,6 @@ static int method_set_x11_keyboard(sd_bus_message *m, void *userdata, sd_bus_err
         if (r < 0)
                 return bus_log_parse_error(r);

-        return sd_bus_error_setf(error, SD_BUS_ERROR_NOT_SUPPORTED,
-            "Changing system settings via systemd is not supported on NixOS.");
-
         x11_context_empty_to_null(&in);

         r = x11_context_verify_and_warn(&in, LOG_ERR, error);
--
2.51.0
